packages/page-settings/src/modals/NetworkSpecs.tsx

```diff
@@ -0,0 +1,180 @@
+// Copyright 2017-2020 @polkadot/app-settings authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { getSystemChainColor } from '@polkadot/apps-config/ui/general';
+import { Button, Input, Modal } from '@polkadot/react-components';
+import { BareProps } from '@polkadot/react-components/types';
+import { useApi, useDebounce } from '@polkadot/react-hooks';
+import registry from '@polkadot/react-api/typeRegistry';
+import ApiPromise from '@polkadot/api/promise';
+import { createType } from '@polkadot/types';
+import addressDefaults from '@polkadot/util-crypto/address/defaults';
+import { QrNetworkSpecs } from '@polkadot/react-qr';
+import { NetworkSpecsStruct } from '@polkadot/ui-settings';
+
+import React, { useEffect, useReducer, useState } from 'react';
+import styled from 'styled-components';
+
+import { useTranslation } from '../translate';
+
+interface Props extends BareProps {
+  onClose: () => void;
+}
+
+function getRandomColor (): string {
+  const letters = '0123456789ABCDEF';
+  let color = '#';
+  for (let i = 0; i < 6; i++) {
+    color += letters[Math.floor(Math.random() * 16)];
+  }
+  return color;
+}
+
+const buildNetworkSpecs = async (api: ApiPromise, systemChain: string, systemName: string): Promise<Partial<NetworkSpecsStruct>> => {
+  const [properties, blockHash] = await Promise.all([
+    api.rpc.system.properties(),
+    api.rpc.chain.getBlockHash(0)
+  ]);
+  const DEFAULT_DECIMALS = createType(registry, 'u32', 12);
+  const DEFAULT_SS58 = createType(registry, 'u32', addressDefaults.prefix);
+  const title = systemChain.toString();
+  const chainColor = getSystemChainColor(systemChain, systemName) || getRandomColor();
+  return {
+    color: chainColor,
+    decimals: properties.tokenDecimals.unwrapOr(DEFAULT_DECIMALS).toNumber(),
+    prefix: properties.ss58Format.unwrapOr(DEFAULT_SS58).toNumber(),
+    unit: properties.tokenSymbol.toString(),
+    title,
+    genesisHash: blockHash.toString()
+  };
+};
+
+function NetworkSpecs ({ className, onClose }: Props): React.ReactElement<Props> | null {
+  const { t } = useTranslation();
+  const initialState = {
+    decimals: 0,
+    prefix: 0,
+    unit: 'UNIT',
+    title: '',
+    color: getRandomColor(),
+    genesisHash: ''
+  };
+  const [qrData, setQrData] = useState<NetworkSpecsStruct>(initialState);
+  const debouncedQrData = useDebounce(qrData, 500);
+  const reducer = (state: NetworkSpecsStruct, delta: Partial<NetworkSpecsStruct>): NetworkSpecsStruct => {
+    const newState = {
+      ...state,
+      ...delta
+    };
+    setQrData(newState);
+    return newState;
+  };
+  const [networkSpecs, setNetworkSpecs] = useReducer(reducer, initialState);
+
+  const { api, systemChain, systemName, isApiReady } = useApi();
+  if (!isApiReady) return null;
```

Comment:
```suggestion
  const { api, systemChain, systemName, isApiReady } = useApi();
  
  if (!isApiReady) {
    return null;
  }
```

Spaceing and no inline returns. Additionally, all returns needs to be below all effects, states and callbacks.